knative-operator/pkg/monitoring/sources/source_deployment_discovery_controller.go

```diff
@@ -53,7 +61,7 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		}
 		return nil
 	})
-	err = c.Watch(&source.Kind{Type: &v1.Deployment{}}, handler.EnqueueRequestsFromMapFunc(enqueueRequests), skipDeletePredicate{}, skipUpdatePredicate{})
+	err = c.Watch(&source.Kind{Type: &v1.Deployment{}}, handler.EnqueueRequestsFromMapFunc(enqueueRequests))
```

Comment:
not really deep into it, but why are the predicates not needed anymore ?

Reply:
Update is needed when deployments are updated due to changes in the observability config in the eventing CR for example which maps to changes in the deployments env vars. Delete can be safely ignored.